content/ideas/algebraic-graphs.md

```diff
@@ -0,0 +1,62 @@
+---
+title: More graph algorithms for Alga
+---
+
+While graphs are ubiquitous in computer programming,
+they have long been underrepresented in Haskell
+due to the imperative nature and lack of type-safety
+of the existing implementations.
+
+The [Algebraic Graphs library](https://github.com/snowleopard/alga) has so far succeded 
+(thanks in part to last years summer of code) in providing a common, 
+typesafe interface for graph instances. However, general ecosystem adoption remains low,
+in part due to the lack of graph algorithms provided.
+
+Therefore the goal of this year's Summer of Code is
+to increase the usefulness of alga for a wider audience:
+
+## Goal 1: Develop a type-safe representation for acyclic graphs in alga.
+
+Acyclic graphs are both common and heavily used in dependency
+management. Improvements in this area would therefore directly
+benefit downstream packages like `build` or `aura`.
+
+In particular, the result should be a type-safe abstraction,
+that makes it easier to work with algorithms like `scc` or `topSort`.
+
+## Goal 2: Implement common graph algorithms like Kruskal, Dijkstra and Moore-Bellman-Ford
+
+Thanks to the edge-labels developed in the build-up of
+last year's Summer of Code, it is now possible to encode
+distances in Alga; however, this feature is rarely used
+as there are nearly zero algorithms provided.
+
+The student should provide algorithms solving the following problems 
+on at least one representation:
```

Comment:
I'd drop "on at least one representation" as it's not immediately clear what this means.